### PR TITLE
ci: panic when log output contains invalid chars (see #589)

### PR DESCRIPTION
**Pull request by altergui**
Signed-off-by: Gui Iribarren <gui@altermundi.net>

### CODE REVIEW

**Comment by altergui** [comment]
(i have no idea what's this "Codeac was not able to perform analysis", new to me)

**Comment by jordipainan** [comment]
Codeac is an external tool for improving code quality, this was introduced months ago, but it is difficult to keep up to date with all warnings and things that "need to be changed".
Lets try to find why it was not able to perform the analysis.

The PR itself lgtm :)

File: log/log.go [hunk 1]
```diff
@@ -109,31 +119,49 @@ func writeErrorToFile(msg string) {
...
 }
 
+// checkInvalidChars checks if the formatted string contains '�' (U+FFFD)
+// and panics if env LOG_PANIC_ON_INVALIDCHARS bool is true.
+//
+// In production (LOG_PANIC_ON_INVALIDCHARS != true), this function returns immediately, no performance hit
+func checkInvalidChars(args ...interface{}) {
+	if panicOnInvalidChars {
+		s := fmt.Sprint(args...)
+		if strings.ContainsRune(s, '�') {
```
Comment (jordipainan) [inline log/log.go]: @altergui seems that Codeac is failing here because it does not recognize this character: `�` and interprets it as `''??'`

Comment (altergui) [reply log/log.go]: haha does codeac come from prehistoric asciiland? that's a perfectly valid UTF-8 char. but ok, i'll escape it , thanks a million for digging into this (clicking on the codeac link prompted me for a login and i stopped there)

Comment (altergui) [reply log/log.go]: amended in https://github.com/vocdoni/vocdoni-node/pull/590/commits/637166ea7d806eccbf9ec199dedea4d36b250f4e

**Comment by altergui** [comment]
now codeac doesn't crash anymore :rofl:

**Review by jordipainan — APPROVED** [review]

**Pull request merged**